src/agent0/core/hyperdrive/interactive/i_hyperdrive.py

```diff
@@ -184,15 +192,18 @@ def _init_agent(
         policy: Type[HyperdriveBasePolicy] | None,
         policy_config: HyperdriveBasePolicy.Config | None,
         private_key: str,
-    ) -> HyperdriveAgent[IHyperdrivePolicy]:
+    ) -> HyperdriveAgent[HyperdriveBasePolicy]:
         # Setting the budget to 0 here, we'll update the wallet from the chain
-        agent = HyperdriveAgent(
-            Account().from_key(private_key),
-            initial_budget=FixedPoint(0),
-            policy=IHyperdrivePolicy(
-                IHyperdrivePolicy.Config(sub_policy=policy, sub_policy_config=policy_config, rng=self.config.rng)
-            ),
-        )
+        if policy is None:
```

Comment:
we do have a NoOp base policy, which I think is a good default if none is provided.

Reply:
There's too many generics down the chain that makes this very clunky to use the base class for this. Keeping this as is.